admin/acceptance.py

```diff
@@ -572,23 +590,20 @@ def main(reactor, args, base_path, top_level):
             runner.stop_nodes(reactor)
         elif options['keep']:
             print "--keep specified, not destroying nodes."
-            print ("To run acceptance tests against these nodes, "
-                   "set the following environment variables: ")
-
-            environment_variables = {
-                'FLOCKER_ACCEPTANCE_NODES':
-                    ':'.join(node.address for node in nodes),
-                'FLOCKER_ACCEPTANCE_CONTROL_NODE': control_node.address,
-                'FLOCKER_ACCEPTANCE_AGENT_NODES':
-                    ':'.join(node.address for node in nodes),
-                'FLOCKER_ACCEPTANCE_VOLUME_BACKEND': dataset_backend.name,
-                'FLOCKER_ACCEPTANCE_API_CERTIFICATES_PATH': ca_directory.path,
-            }
-
-            for environment_variable in environment_variables:
-                print "export {name}={value};".format(
-                    name=environment_variable,
-                    value=environment_variables[environment_variable],
-                )
+            try:
+                cluster
+            except NameError:
```

Comment:
Using NameError or AttributeError like this is a recipe for bugs, since it can't distinguish typos from expected variable. Instead you can do:

```
cluster = None
try:
    cluster = yield runner.start_cluster()
    ....
finally:
    if cluster is not None: ...
```